hack/generated/pkg/testcommon/ensure.go

```diff
@@ -31,8 +31,8 @@ func NewEnsure(c client.Client, stateAnnotation string, errorAnnotation string)
 	}
 }
 
-// Provisioned checks to ensure the provisioning state of the resource is successful.
-func (e *Ensure) Provisioned(ctx context.Context, obj runtime.Object) (bool, error) {
+// State checks to ensure the provisioning state of the resource the target state.
```

Comment:
```suggestion
// State checks to check whether the provisioning state of the resource has the target state.
```

"Ensure" is an active word that implies this method will make any changes required for the resource to be in target state; I don't think that's what you mean.

Reply:
Maybe call the method `HasState()`?

Reply:
> "Ensure" is an active word that implies this method will make any changes required for the resource to be in target state; I don't think that's what you mean.

The actual struct is already called `Ensure` - but the methods hanging off as you note do not actually make changes, they just check them. This same comment style is repeated for the other methods `Provisioned`, `Failed`, etc.

[Merriam-Webster](https://www.merriam-webster.com/dictionary/ensure) says:
> Ensure: to make sure, certain, or safe : guarantee

whereas [Cambridge dictionary](https://dictionary.cambridge.org/us/dictionary/english/ensure) agrees with you:
> To make something certain to happen

I'm going to hold off making a change on this - if we do want to change it we should change the whole class, not just this one place.

Reply:
Changed to `HasState`

Reply:
I made a note to change Ensure to something else, maybe `Verify`? Will do it when I have some spare time.